### ISSUE TITLE
manager-based build seems to be broken on OS X

### ISSUE THREAD

**Issue by wdanilo** [luna/luna-manager#155]
@binary132 commented on [Sat Feb 10 2018](https://github.com/luna/luna-studio/issues/813)

This is built against my upstream which has the S3 links corrected to use packages.luna-lang.org: https://github.com/luna/luna-studio/pull/812

```sh
luna-manager develop luna-studio --path "${HOME}/meta/dev/workshop/workspace" > results.txt 2> errors.txt
```

> results.txt (had already built and run with same errors, so git clone / etc weren't needed.)

http://pastebin.centos.org/543396/

> errors.txt

http://pastebin.centos.org/543381/

---

@wdanilo commented on [Sat Feb 10 2018](https://github.com/luna/luna-studio/issues/813#issuecomment-364700004)

This error is very strange. I have never seen it before. It's clearly Haskell's stack error. 

Ok, after some minor investigation (https://stackoverflow.com/questions/48080510/stack-setup-error-solver-must-be-one-of-modular) my first bet is that you've got Cabal 2.0 isntalled system wide. Stack doesnt support Cabal 2.0 yet. However, if that's it, this is our bug that we allow to use systemwide Cabal installation instead of sandboxed one. What does `cabal --version` prints in your terminial?

---

@wdanilo commented on [Sat Feb 10 2018](https://github.com/luna/luna-studio/issues/813#issuecomment-364700169)

Btw, wrong repo. Moving to luna-manager.

**Comment by binary132** [comment]
Good call, it's a dependency of the system haskell.  You'd think that `stack` wouldn't try to use the global by default.... maybe adjusting my PATH can help fix this.

```
~❯ cabal --version
cabal-install version 2.0.0.1
compiled using version 2.0.1.0 of the Cabal library

~❯ brew info cabal-install
cabal-install: stable 2.0.0.1 (bottled), HEAD
```

^ cabal-install is a dependency of something else in homebrew, btw.

**Comment by wdanilo** [comment]
adjusting your path should work as a hot fix. This bug have to be eliminsated in luna-manager too

**Comment by binary132** [comment]
Confirmed, moving `cabal` and `ghc` temporarily away from my PATH allowed `luna-manager develop` to succeed.

**Comment by binary132** [comment]
I suspect that for some reason the Shelly context does not inherit the environment variables correctly.  I'll see if I can set them with `Environment.setEnv` instead of sourcing the shell script...

**Comment by wdanilo** [comment]
What? Moving `cabal` and `ghc` from your path resolved it? Hmm, then it seems like our path has lower priority than your system one. It's worth checking how the scripts append our path to $PATH and be sure they append it in front. If this is caused by such stupid thing like appending path to the end, it will also explain why libs are taken from wrong place - just thinking, I dont know this code completely and cannot look into it atm.

**Comment by wdanilo** [comment]
@binary132 ok, looking at the code everytihng seems ok. If you have a little time to track it down, the line which prepends everythign to path is here: https://github.com/luna/luna-manager/blob/master/luna-manager/src/Luna/Manager/Command/Develop.hs#L114 

I'm wondering why it doesnt work for you. It works for us, so it will be great if you try to check what happens there on your side (printing $PATH etc is good enough to check if thats correct)

**Comment by binary132** [comment]
Yeah, it is prepended.  When I try to print `os.environ["LD_LIBRARY_PATH"]` in `scripts_build/build.py`, I get `raise KeyError(key) from None`.  Trying to reproduce in a simpler case to see why this is happening....

**Comment by wdanilo** [comment]
interesting. Thank you for working on it!

**Comment by binary132** [comment]
During the `gyp install` step:

```
gyp ERR! stack Error: Python executable "/Users/bodie/meta/dev/workshop/workspace/luna-develop/apps/luna-studio/dist/third-party/Atom.app/Contents/Resources/app/apm/bin/python-interceptor.sh" is v3.6.2, which
 is not supported by gyp.
```

**Comment by binary132** [comment]
https://stackoverflow.com/a/856287 might be relevant...

**Comment by wdanilo** [comment]
Actually I'm sure we've been discussing `gyp` some time aga and its working here. Im sure @sylwiabr will know more here.

**Comment by binary132** [comment]
https://ghc.haskell.org/trac/ghc/ticket/11617#comment:1

**Comment by binary132** [comment]
I'll open an issue specifically to track and resolve the `LD_LIBRARY_PATH` item.  I'll leave any further notes on unrelated items in here.....

**Comment by binary132** [comment]
I can't reproduce the `cabal` conflict, maybe it was due to a partial build or partial download?  The remaining issues are:

 - `gyp install` fails (known issue, work around with --release)
 - `libzmq` not found (known issue, work around with `brew install zeromq`)
 - GHCJS build step fails frequently with heap exhaustion (work around by retrying)
